include/actuator/ActuatorTypes.h

```diff
@@ -0,0 +1,86 @@
+// (NTESS), National Renewable Energy Laboratory, University of Texas Austin,
+// Northwest Research Associates. Under the terms of Contract DE-NA0003525
+// with NTESS, the U.S. Government retains certain rights in this software.
+//
+// This software is released under the BSD 3-clause license. See LICENSE file
+// for more details.
+//
+
+#ifndef INCLUDE_ACTUATOR_ACTUATORTYPES_H_
+#define INCLUDE_ACTUATOR_ACTUATORTYPES_H_
+
+#include <Kokkos_Core.hpp>
+#include <Kokkos_DualView.hpp>
+
+namespace sierra {
+namespace nalu {
+
+#ifdef ACTUATOR_ON_DEVICE
+using ActuatorMemSpace = Kokkos::CudaSpace;
+using ActuatorMemLayout = Kokkos::LayoutLeft;
+using ActuatorExecutionSpace = Kokkos::DefaultExecutionSpace;
+#else
+using ActuatorMemSpace = Kokkos::HostSpace;
+using ActuatorMemLayout = Kokkos::LayoutRight;
+using ActuatorExecutionSpace = Kokkos::DefaultHostExecutionSpace;
+#endif
+using ActuatorFixedMemSpace = Kokkos::HostSpace;
+using ActuatorFixedMemLayout = Kokkos::LayoutRight;
+using ActuatorFixedExecutionSpace = Kokkos::DefaultHostExecutionSpace;
+
+// DUAL VIEWS
+using ActScalarIntDv =
+  Kokkos::DualView<int*, ActuatorMemLayout, ActuatorMemSpace>;
+using ActScalarU64Dv =
+  Kokkos::DualView<uint64_t*, ActuatorMemLayout, ActuatorMemSpace>;
+using ActScalarDblDv =
+  Kokkos::DualView<double*, ActuatorMemLayout, ActuatorMemSpace>;
+using ActVectorDblDv =
+  Kokkos::DualView<double* [3], ActuatorMemLayout, ActuatorMemSpace>;
+
+// VIEWS
+using ActScalarInt = Kokkos::View<int*, ActuatorMemLayout, ActuatorMemSpace>;
+using ActScalarU64 =
+  Kokkos::View<uint64_t*, ActuatorMemLayout, ActuatorMemSpace>;
+using ActScalarDbl = Kokkos::View<double*, ActuatorMemLayout, ActuatorMemSpace>;
+using ActVectorDbl =
+  Kokkos::View<double* [3], ActuatorMemLayout, ActuatorMemSpace>;
+
+// VIEWS FIXED TO HOST
+using ActFixRangePolicy = Kokkos::RangePolicy<ActuatorFixedExecutionSpace>;
+using ActFixScalarInt =
+  Kokkos::View<int*, ActuatorFixedMemLayout, ActuatorFixedMemSpace>;
+using ActFixScalarDbl =
+  Kokkos::View<double*, ActuatorFixedMemLayout, ActuatorFixedMemSpace>;
+using ActFixVectorDbl =
+  Kokkos::View<double* [3], ActuatorFixedMemLayout, ActuatorFixedMemSpace>;
+using ActFixElemIds =
+  Kokkos::View<uint64_t*, ActuatorFixedMemLayout, ActuatorFixedMemSpace>;
+using ActFixScalarBool =
+  Kokkos::View<bool*, ActuatorFixedMemLayout, ActuatorFixedMemSpace>;
+using ActFixArrayInt =
+  Kokkos::View<int**, ActuatorFixedMemLayout, ActuatorFixedMemSpace>;
+
+template <typename memory_space>
+struct ActDualViewHelper
+{
+  template <typename T>
+  KOKKOS_INLINE_FUNCTION auto get_local_view(T dualView) const
+    -> decltype(dualView.template view<memory_space>())
+  {
+    return dualView.template view<memory_space>();
+  }
+
+  template <typename T>
+  KOKKOS_INLINE_FUNCTION void touch_dual_view(T dualView)
+  {
+    dualView.template sync<memory_space>();
+    dualView.template modify<memory_space>();
+  }
+};
```

Comment:
It is used in ‘ActuatorFunctorFAST.h’ and some unit tests. It is for functors that would use the non ‘Fixed’ execution space where it can end up running on host or device depending on how the code is compiled.